src/fparser/two/utils.py

```diff
@@ -1500,64 +1594,74 @@ def tostr(self):
             return '%s, %s :: %s' % self.items
 
 
-def walk_ast(children, my_types=None, indent=0, debug=False):
+def walk(node_list, types=None, indent=0, debug=False):
     '''
-    Walk down the tree produced by fparser2 where children
-    are listed under 'content'.  Returns a list of all nodes with the
-    specified type(s).
-
-    :param children: list of child nodes from which to walk.
-    :type children: list of :py:class:fparser.two.utils.Base.
-    :param my_types: list of types of Node to return. (Default is to \
-                     return all nodes.)
-    :type my_types: list of type
+    Walk down the parse tree produced by fparser2.  Returns a list of all
+    nodes with the specified type(s).
+
+    :param node_list: node or list of nodes from which to walk.
+    :type node_list: (list of) :py:class:fparser.two.utils.Base
+    :param types: type or tuple of types of Node to return. (Default is to \
+                  return all nodes.)
+    :type types: type or tuple of types
     :param int indent: extent to which to indent debug output.
     :param bool debug: whether or not to write textual representation of AST \
                        to stdout.
     :returns: a list of nodes
     :rtype: `list` of :py:class:`fparser.two.utils.Base`
     '''
     local_list = []
-    for child in children:
+
+    if not isinstance(node_list, (list, tuple)):
+        node_list = [node_list]
+
+    for child in node_list:
         if debug:
             if isinstance(child, str):
                 print(indent*"  " + "child type = ", type(child), repr(child))
             else:
                 print(indent*"  " + "child type = ", type(child))
-        if my_types is None or type(child) in my_types:
+        if types is None or isinstance(child, types):
             local_list.append(child)
-
-        # Depending on their level in the tree produced by fparser2003,
-        # some nodes have children listed in .content and some have them
-        # listed under .items. If a node has neither then it has no
-        # children.
-        if hasattr(child, "content"):
-            local_list += walk_ast(child.content, my_types, indent+1, debug)
-        elif hasattr(child, "items"):
-            local_list += walk_ast(child.items, my_types, indent+1, debug)
+        # Recurse down
+        local_list += walk(get_children(child), types, indent+1, debug)
 
     return local_list
 
 
-def get_child(root_node, node_type):
+def get_children(node):
```

Comment:
As discussed, please replace this with a children property/attribute in the `Base` base class.

Reply:
Done.